lib/jmespath/nodes/condition.rb

```diff
@@ -28,8 +28,8 @@ def optimize
 
     class ComparatorCondition < Node
       COMPARATOR_TO_CONDITION = {}
-      COMPARABLE_TYPES = [Integer, String].freeze
-
+      COMPARABLE_TYPES = [Integer, Float, String].freeze
+      NUMERIC_TYPES = [Integer, Float].freeze
```

Comment:
Why not using Numeric, that is parent class for Float and BigDecimal?

Reply:
Because json gets parsed into Float's not numeric.... I am using `left_value.class` not `left_value.is_a?`

Reply:
Just curious... what is the purpose of putting String there?

Reply:
Updated

Reply:
I'm using Float in the class vars because when parsing json it will be type float not numeric

Reply:
When using `COMPARABLE_TYPES.include?` then I agree, that you cannot use Numeric and I think Float is working with the most cases.

Reply:
>When using COMPARABLE_TYPES.include? then I agree, that you cannot use Numeric and I think Float is working with the most cases.

I'm confused. Why not use Numeric here?